actors/miner/src/lib.rs

```diff
@@ -4935,6 +5052,87 @@ struct ReplicaUpdateActivatedData {
     verified_space: BigInt,
 }
 
+fn activate_pieces(
+    rt: &impl Runtime,
+    activation_inputs: Vec<PiecesActivationInput>,
+    all_or_nothing: bool,
+) -> Result<(BatchReturn, Vec<DataActivationOutput>), ActorError> {
+    // Get a flattened list of verified claims for all activated sectors
+    let mut verified_claims = Vec::new();
+    let mut computed_commds = Vec::new();
+    let mut unverified_spaces = Vec::new();
+    for activation_info in activation_inputs {
+        if !activation_info.pieces.is_empty() {
+            let mut pieces = vec![];
+            for piece_activation in sector_activation.pieces.iter() {
+                pieces.push(PieceInfo { cid: piece.cid, size: piece.size });
+            }
+            let computed_commd = rt.compute_unsealed_sector_cid(activation_info.sector_type, &pieces).map_err(|e| {
+                e.downcast_default(ExitCode::USR_ILLEGAL_ARGUMENT, "failed to compute unsealed sector CID")?;
+            });
+            // If a CommD was declared, check it matches.
```

Comment:
I don't think it's possible to be in this path if CommD wasn't declared.

Reply:
Not as called today but I was thinking it generally could be to match structure of the deals data activation. 

But it should be fine to explicitly keep these two activation code paths separate and make CommD declaration as a precondition to activate_pieces.